src/shogun/kernel/string/WeightedDegreePositionStringKernel.cpp

```diff
@@ -1877,90 +1877,98 @@ void CWeightedDegreePositionStringKernel::load_serializable_post() throw (Shogun
-			"Weights per position.");
+	m_parameters->add_vector(
+	    &position_weights, &position_weights_len, "position_weights",
+	    "Weights per position.");
 	watch_param("position_weights", &position_weights, &position_weights_len);
 
-	m_parameters->add_vector(&position_weights_lhs, &position_weights_lhs_len,
-			"position_weights_lhs",
-			"Weights per position left hand side.");
+	m_parameters->add_vector(
+	    &position_weights_lhs, &position_weights_lhs_len,
+	    "position_weights_lhs", "Weights per position left hand side.");
 	watch_param(
 	    "position_weights_lhs", &position_weights_lhs,
 	    &position_weights_lhs_len);
 
-	m_parameters->add_vector(&position_weights_rhs, &position_weights_rhs_len,
-			"position_weights_rhs",
-			"Weights per position right hand side.");
+	m_parameters->add_vector(
+	    &position_weights_rhs, &position_weights_rhs_len,
+	    "position_weights_rhs", "Weights per position right hand side.");
 	watch_param(
 	    "position_weights_rhs", &position_weights_rhs,
 	    &position_weights_rhs_len);
 
-	m_parameters->add_vector(&shift, &shift_len,
-			"shift",
-			"Shift Vector.");
+	m_parameters->add_vector(&shift, &shift_len, "shift", "Shift Vector.");
 	watch_param("shift", &shift, &shift_len);
 
-	SG_ADD(&max_shift, "max_shift", "Maximal shift.", ParameterProperties::HYPER);
-	SG_ADD(&mkl_stepsize, "mkl_stepsize", "MKL step size.", ParameterProperties::HYPER);
-	SG_ADD(&degree, "degree", "Order of WD kernel.", ParameterProperties::HYPER);
-	SG_ADD(&max_mismatch, "max_mismatch",
-			"Number of allowed mismatches.", ParameterProperties::HYPER);
-	SG_ADD(&block_computation, "block_computation",
-			"If block computation shall be used.");
-	SG_ADD((machine_int_t*) &type, "type",
-			"WeightedDegree kernel type.", ParameterProperties::HYPER);
-	SG_ADD(&which_degree, "which_degree",
-			"The selected degree. All degrees are used by default (for value -1).",
-			ParameterProperties::HYPER);
-	SG_ADD((CSGObject**) &alphabet, "alphabet",
-			"Alphabet of Features.");
+	SG_ADD(
+	    &max_shift, "max_shift", "Maximal shift.", ParameterProperties::HYPER);
+	SG_ADD(
+	    &mkl_stepsize, "mkl_stepsize", "MKL step size.",
+	    ParameterProperties::HYPER);
+	SG_ADD(
+	    &degree, "degree", "Order of WD kernel.", ParameterProperties::HYPER);
+	SG_ADD(
+	    &max_mismatch, "max_mismatch", "Number of allowed mismatches.",
+	    ParameterProperties::HYPER);
+	SG_ADD(
+	    &block_computation, "block_computation",
+	    "If block computation shall be used.");
+	SG_ADD(
+	    &which_degree, "which_degree",
+	    "The selected degree. All degrees are used by default (for value -1).",
+	    ParameterProperties::HYPER);
+	SG_ADD((CSGObject**)&alphabet, "alphabet", "Alphabet of Features.");
+	SG_ADD_OPTIONS(
+	    (machine_int_t*)&type, "type", "WeightedDegree kernel type.",
+	    ParameterProperties::HYPER, E_WD, E_EXTERNAL, E_BLOCK_CONST,
+	    E_BLOCK_LINEAR, E_BLOCK_SQPOLY, E_BLOCK_CUBICPOLY, E_BLOCK_EXP,
```

Comment:
Could it be something like `SG_OPTIONS(E_EXTERNAL, E_BLOCK_CONST, E_BLOCK_LINEAR, E_BLOCK_SQPOLY, E_BLOCK_CUBICPOLY, E_BLOCK_EXP)` for readability?